### ISSUE TITLE
Created files should have the same time as the original file.

### ISSUE THREAD

**Issue by cliveontoast** [cliveontoast/GoMoPho#13]
The movie file has the time/date of the moment it is being saved instead original date of creation.

It would be better to have the date of the original file.

This is especially handy when you combine video's and live-video's, then you can sort by date of creation. Searching through 800 files will be easier then :)

Supplied by @stefansegers

**Comment by cliveontoast** [comment]
Will need to revisit this. The date taken is a bit harder to get. But looks like it's available easily in .net core 3.0

property item 36867

This also might make the "where's the video start" much easier .... considering that's in the EXIF data

https://stackoverflow.com/questions/180030/how-can-i-find-out-when-a-picture-was-actually-taken-in-c-sharp-running-on-vista

https://docs.microsoft.com/en-us/dotnet/api/system.drawing.imaging.propertyitem.id?redirectedfrom=MSDN&view=netframework-4.8#System_Drawing_Imaging_PropertyItem_Id

**Comment by cliveontoast** [comment]
Bit of a wait between beers. The issue is resolved, grab the new version 3.1 or higher and give it a go. Let me know how you go.

https://github.com/cliveontoast/GoMoPho/releases

**Comment by jacobgkau** [comment]
This doesn't seem to be working anymore. With the Linux build, I'm seeing the timestamp for the MP4 file set to the time I ran the command, not the time the image file was created.

**Comment by cliveontoast** [comment]
TLDR:
You're right @jacobgkau 

Looks like its never worked on my Ubuntu WSL instance either. I've tried version 3.1 and 3.7.

On Debian-based Linux distributions you can use apt-get to install the dependencies:
```
sudo apt-get install libgdiplus
```

Long story for my records:

I receive the following error

```
Did not correct file date of test-image/MVIMG_20180910_124410.jpg The type initializer for 'Gdip' threw an exception.
```

googling... 
https://www.mono-project.com/docs/gui/libgdiplus/
https://github.com/mono/libgdiplus <-- This repository was archived by the owner on Mar 8, 2025. It is now read-only. 

So that's a bit of a worry.

more reading.

https://lxadm.com/the-type-initializer-for-gdip-threw-an-exception/
https://stackoverflow.com/questions/67091719/system-drawing-common-the-type-initializer-for-gdip-threw-an-exception
https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only
https://github.com/dotnet/runtime/issues/27200
